cli/boards.go

```diff
@@ -362,8 +362,17 @@ func createArchive(file *os.File) (*bytes.Buffer, error) {
 }
 
 // helper function to download a url to a local file.
-func downloadFile(ctx context.Context, filepath, url string) error {
+func (c *viamClient) downloadFile(ctx context.Context, filepath, url string) error {
 	getReq, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+
+	// HTTP Requests to app require auth headers, varies by auth mechanism.
+	if token, isToken := c.conf.Auth.(*token); isToken {
+		getReq.Header.Set("Authorization", "Bearer "+token.AccessToken)
+	} else if APIKey, isAPIKey := c.conf.Auth.(*apiKey); isAPIKey {
+		getReq.Header.Set("key_id", APIKey.KeyID)
+		getReq.Header.Set("key", APIKey.KeyCrypto)
+	}
```

Comment:
makes sense to me! Might be worth pulling out into a method somewhere that can be called from the go commons utilities if there is such a thing?

Reply:
Agreed! I'll do this now. I need todo it anyways to add tests for this logic.